plugins/services/src/js/components/modals/ServiceRootGroupModal/index.tsx

```diff
@@ -0,0 +1,431 @@
+                      name.
+                    </Trans>
+                  ]
+                },
+                isPending: false
+              });
+              return;
+            case "Forbidden - Group":
+              this.setState({
+                errors: {
+                  form: [
+                    <Trans key="groupPermission">
+                      You do not have permission to create a group.
+                    </Trans>
+                  ]
+                },
+                isPending: false
+              });
+              return;
+            default:
+              this.setState({
+                errors: {
+                  form: [
+                    <Trans key="miscGroup">
+                      Unable to create group: {e.message}
+                    </Trans>
+                  ]
+                },
+                isPending: false
+              });
+              return;
+          }
+        }
+      });
+  }
+
+  handleAdvancedSectionClick() {
+    this.setState({
+      expandAdvancedSettings: !this.state.expandAdvancedSettings
+    });
+  }
+
+  getModalContent() {
+    const { data, errors } = this.state;
+    return (
+      <form className="container" onChange={this.handleFormChange}>
+        <ErrorsPanel errors={this.state.errors.form} />
+        <Trans render="h1" className="flush-top short-bottom">
+          General
+        </Trans>
+        <FormRow>
+          <FormGroup
+            className="column-12 column-medium-4"
+            showError={Boolean(errors.id)}
+          >
+            <FieldLabel>
+              <FormGroupHeading required={true}>
+                <Trans render={<FormGroupHeadingContent primary={true} />}>
+                  Name
+                </Trans>
+              </FormGroupHeading>
+            </FieldLabel>
+            <FieldAutofocus>
+              <FieldInput
+                name="id"
+                type="text"
+                value={data.id}
+                disabled={this.props.isEdit}
+              />
+            </FieldAutofocus>
+            <FieldError>{errors.id}</FieldError>
+          </FormGroup>
+        </FormRow>
+        <FormRow>
+          <FormGroup className="column-12">
+            <FieldLabel>
+              <FormGroupHeading>
+                <Trans render={<FormGroupHeadingContent primary={false} />}>
+                  Path
+                </Trans>
+              </FormGroupHeading>
+            </FieldLabel>
+            <div>{getPathFromGroupId(data.id)}</div>
+          </FormGroup>
+        </FormRow>
+        <h2 className="short-bottom">
+          <FormGroupHeading>
+            <Trans render={<FormGroupHeadingContent primary={true} />}>
+              Quota
+            </Trans>
+          </FormGroupHeading>
+        </h2>
+        <Trans render="p">
+          Define the maximum amount of resources that can be used by services in
+          this group.
+        </Trans>
+        <FormRow>
+          <FormGroup
```

Comment:
@TattdCodeMonkey , what do you think about creating these form groups with a function? They are pretty similar.

Reply:
I'd be fine with that. We will likely eventually have to abstract this whole section into its own component.